src/com/amazon/ion/impl/IonTypeID.java

```diff
@@ -136,12 +172,177 @@ private IonTypeID(byte id, int minorVersion) {
+                       // Symbol values with inline text.
+                    || upperNibble == 0x9;
+                IonType typeFromUpperNibble = BINARY_TOKEN_TYPES_1_1[upperNibble];
+                if (typeFromUpperNibble == null) {
+                    if (!isValid) {
+                        type = null;
+                    } else if (upperNibble == 0x5) {
+                        if (lowerNibble <= 0x8) {
+                            type = IonType.INT;
+                            length = lowerNibble;
+                        } else if (lowerNibble >= 0xE) {
+                            type = IonType.BOOL;
+                            length = 0;
+                        } else {
+                            type = IonType.FLOAT;
+                            if (lowerNibble == 0xA) {
+                                length = 0; // 0e0
+                            } else if (lowerNibble == 0xB) {
+                                length = 2;
+                            } else if (lowerNibble == 0xC) {
+                                length = 4;
+                            } else if (lowerNibble == 0xD) {
+                                length = 8;
+                            }
+                        }
+                    } else if (upperNibble == 0xE) {
+                        if (lowerNibble <= 0x3) {
+                            type = IonType.SYMBOL;
+                            length = lowerNibble == 0x3 ? -1 : lowerNibble;
+                        } else if (lowerNibble <= 0x9) {
+                            type = ION_TYPE_ANNOTATION_WRAPPER;
+                        } else if (lowerNibble == 0xA) {
+                            // null.null
+                            type = IonType.NULL;
+                            isNull = true;
+                            length = 0;
+                        } else if (lowerNibble == 0xB) {
+                            // Typed null. Type byte follows.
+                            type = null;
+                            isNull = true;
+                        } else if (lowerNibble <= 0xD) {
+                            isNopPad = true;
+                            type = null;
+                            length = variableLength ? -1 : 0;
+                        } else { // 0xF
+                            // System macro invocation.
+                            type = null;
+                        }
+                    } else { // 0xF
+                        if (lowerNibble == 0) {
+                            // Delimited end
+                            type = null;
+                            length = 0;
+                        } else if (lowerNibble == 0x3 || lowerNibble == 0xC || lowerNibble == 0xD) {
+                            type = IonType.STRUCT;
+                        } else if (lowerNibble == 0x5) {
+                            type = IonType.INT;
+                        } else if (lowerNibble == 0x6) {
```

Comment:
There's an `OpCodes` class, so you could write:
```suggestion
                        } else if (id == OpCodes.VARIABLE_LENGTH_DECIMAL) {
```
...and similarly for others where it's a single value. I don't think it's any more expensive to compare a whole byte instead of a nibble, so it would be nice to do this for readability.

Reply:
Sure, #645